ui/src/app/edge/live/chpsoc/chpsoc.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, Input, Output } from '@angular/core';
+import { ChannelAddress, Edge, EdgeConfig, Service, Websocket } from '../../../shared/shared';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+    selector: ChpsocComponent.SELECTOR,
+    templateUrl: './chpsoc.component.html'
+})
+export class ChpsocComponent {
+
+    private static readonly SELECTOR = "chpSoc";
+
+    @Input() private componentId: string;
+
+    public edge: Edge = null;
+    public controller: EdgeConfig.Component = null;
+    public inputChannel: ChannelAddress = null;
+    public outputChannel: ChannelAddress = null;
+    public lowThreshold: number;
+    public highThresold: number;
```

Comment:
Do you need all of these? Do all of them need to be public? - for example: I do not see 'controller' being used.